**1. Calculate Area of a Circle**

Write a Python program which accepts the radius of a circle from the user and compute the area.

In [17]:
radius = float(input("Enter radius of circle: "))
area = 3.141592654 * radius ** 2
print("Area of Cirlce with radius"+str(radius) +" is "+str(area))

Enter radius of circle: 0.5
Area of Cirlce with radius0.5 is 0.7853981635


**2.	Check Number either positive, negative or zero**

Write a Python program to check if a number is positive, negative or zero

In [21]:
number = int(input("Enter number: "))

if number < 0:
    print("Number is negative.")
elif number > 0:
    print("Number is positive.")
elif number == 0:
    print("Number is Zero.")
else:
    print("Invalid Number.")


Enter number: 0
Number is Zero.


**3.	Divisibility Check of two numbers**


Write a Python program to check whether a number is completely divisible by another number. Accept two integer values form the user

In [24]:
numerator = int(input("Enter Numerator: "))
denominator = int(input("Enter Denominator: "))

if numerator % denominator == 0:
    print(str(numerator)+ " is completely divisible by "+str(denominator))
else:
    print(str(numerator)+ " is not completely divisible by "+str(denominator))


Enter Numerator: 4
Enter Denominator: 2
4 is completely divisible by 2


**4.	Days Calculator**

Write a Python program to calculate number of days between two dates

In [34]:
from datetime import datetime
date_format = "%d/%m/%Y"

date_1 = input("Enter a date in (dd/mm/yy): ")
date_2 = input("Enter a date in (dd/mm/yy): ")
a = datetime.strptime(date_1,date_format)
b = datetime.strptime(date_2,date_format)
delta = b - a
print("There are",delta.days,"days in between",date_1,"and",date_2)

Enter a date in (dd/mm/yy): 12/11/2018
Enter a date in (dd/mm/yy): 19/11/2019
There are 372 days in between 12/11/2018 and 19/11/2019


**5.	Calculate Volume of a sphere**

Write a Python program to get the volume of a sphere, please take the radius as input from user

In [37]:
radius_of_sphere = int(input("Enter radius of sphere: "))
volume_of_sphere = 4/3 * 3.142 * radius_of_sphere ** 3
print("Volume of the sphere with Radius",radius_of_sphere,"is",volume_of_sphere)

Enter radius of sphere: 1
Volume of the sphere with Radius 1 is 4.189333333333333


**6.	Copy string n times**

Write a Python program to get a string which is n (non-negative integer) copies of a given string. 

In [39]:
string = input("Enter String: ")
copies = int(input("How many copies of string you need: "))
print(str(copies) + " Copies of "+string+" are "+string * copies)

Enter String: hi
How many copies of string you need: 10
10 Copies of hi are hihihihihihihihihihi


**7.	Check if number is Even or Odd**

Write a Python program to find whether a given number (accept from the user) is even or odd, print out an appropriate message to the user

In [43]:
number = int(input("Enter Number: "))

if number % 2 == 0:
    print(number,"is even")
else:
    print(number,"is odd")

Enter Number: 8
8 is even


**8.	Vowel Tester**

Write a Python program to test whether a passed letter is a vowel or not

In [5]:
letter = input("Enter Letter: ")
c = letter.lower()
if c == "a" or c == "e" or c == "i" or c == "o" or c == "u":
    print("Letter",letter,"is vowel.")
else:
    print("Letter",letter,"is Not vowel.")

Enter Letter: N
Letter N is Not vowel.


**9.	Triangle area**

Write a Python program that will accept the base and height of a triangle and compute the area

In [8]:
t_base = int(input("Enter magnitude of triangle base"))
t_height = int(input("Enter magnitude of triangle height"))
area_of_triangle = t_base * t_height //2
print("Area of a Triangle with Height", t_height,"and Base",t_base,"is",area_of_triangle)

Enter magnitude of triangle base10
Enter magnitude of triangle height4
Area of a Triangle with Height 4 and Base 10 is 20


**10.	Calculate Interest**

Write a Python program to compute the future value of a specified principal amount, rate of interest, and a number of years

In [14]:
p = int(input("Enter principle amount: "))
i = float(input("Enter rate of interest in %: "))
t = int(input("Enter number of years of investment: "))
r = p*((1+(i)) ** t)
print("After",t,"years your principle amount",p,"over an interest of",i,"% will be",r)

Enter principle amount: 10000
Enter rate of interest in %: 0.1
Enter number of years of investment: 5
After 5 years your principle amount 10000 over an interest of 0.1 % will be 16105.100000000006


**11.	Euclidean distance**

Write a Python program to compute the distance between the points (x1, y1) and (x2, y2).

In [22]:
import math
x1 = int(input("Enter Co-ordinate for x1: "))
x2 = int(input("Enter Co-ordinate for x2: "))
y1 = int(input("Enter Co-ordinate for y1: "))
y2 = int(input("Enter Co-ordinate for y2: "))

distance = (x1 - x2)*(x1 - x2) + (y1 - y2)*(y1 - y2)
e = math.sqrt(distance)
print("Distance between Points",(x1,x2),"and",(y1,y2),"is",e)

Enter Co-ordinate for x1: 10
Enter Co-ordinate for x2: 9
Enter Co-ordinate for y1: 2
Enter Co-ordinate for y2: 1
Distance between Points (10, 9) and (2, 1) is 1.4142135623730951


**12.	Feet to Centimeter Converter**

Write a Python program to convert height in feet to centimetres.

In [25]:
height = int(input("Enter Height in Feet: "))
d = height * 30.48
print("There are",d,"Cm in",height,"ft")

Enter Height in Feet: 5
There are 152.4 Cm in 5 ft


**13.	BMI Calculator**

Write a Python program to calculate body mass index

In [41]:
h = int(input("Enter Height in Cm: "))
w = int(input("Enter Weight in kg: "))

bmi = (w / (h * 0.01) ** 2)
print("Your BMI is",bmi)

Enter Height in Cm: 180
Enter Weight in kg: 75
Your BMI is 23.148148148148145


**14.	Sum of n Positive Integers**

Write a python program to sum of the first n positive integers

In [46]:
n = int(input("Enter a value of n: "))
total = n * (n+1)//2
print("Sum of n Postive integers till",n,"is",total)

Enter a value of n: 4
Sum of n Postive integers till 4 is 10


**15.	Digits Sum of a Number**

Write a Python program to calculate the sum of the digits in an integer

In [67]:
n = input("Enter a number: ")

total = 0
for number in n:
    total += int(number)
   

print("Sum of",'+'.join(n),"is",total)


Enter a number: 1234
Sum of 1+2+3+4 is 10


**16.	Decimal to Binary Converter**

Write a Python program to convert an decimal integer to binary


In [3]:
dec = int(input("Enter Decimal number: "))
binary = bin(dec)
print("Binary Representation of",dec,"is",binary[2:])

Enter Decimal number: 32
Binary Representation of 32 is 100000


**17.	Binary to Decimal Converter**

Write a Python program to convert an binary number to decimal number


In [12]:
binary = input("Enter binary number: ")
dec = int(binary,2)
print("Decimal Reperesentation of",binary,"is: ",dec)

Enter binary number: 1001
Decimal Reperesentation of 1001 is:  9


**18.	Vowel and Constant Conunter**

Input a text and count the occurences of vowels and consonent


In [27]:
text = input("Enter text: ")
text = text.lower()
vowels = 0
consonent = 0

for elem in text:
    if elem == "a" or elem == "e" or elem == "i" or elem == "o" or elem == "u":
        vowels +=1
    else:
        consonent +=1

print("Vowels: ",vowels)
print("Consonent: ",consonent)

Enter text: QuickBrownFoxJumpsovertheDog
Vowels:  9
Consonent:  19


**19.	Palindrome Tester**

write a program to check wheter given input is palindrome or not


In [31]:
text_check = input("Enter text: ")

reverse_text = text_check[::-1]

if text_check == reverse_text:
    print("Text",text_check,"is palindrome")
else:
    print("Text ",text_check,"is not palindrome")


Enter text: AHA
Text AHA is palindrome


**20.	Count Aplhabets,Numbers and special characters**

write a python program that accepts a string and calculate the number of digits and letters


In [64]:
characters = input("Enter text: ")
numbers = 0
alphabets = 0
special_characters = 0
spaces = 0

for elem in characters:
    if elem.isalpha():
        alphabets += 1
    elif elem.isnumeric():
        numbers +=1
    elif elem.isspace():
        spaces +=1
    elif elem.isascii():
        special_characters += 1

print("Numbers: ",numbers)
print("Alphabets: ",alphabets)
print("Special Characters: ",special_characters)
print("Spaces: ",spaces)


Enter text: python 3.7
Numbers:  2
Alphabets:  6
Special Characters:  1
Spaces:  1


**21.	Count Aplhabets,Numbers and special characters**

write a python program that accepts a string and calculate the number of digits and letters

    *
    **
    ***
    ****
    *****
    ****
    ***
    **
    *

In [85]:
for i in range(0,6):
    for j in range(i):
        print("*",end=" ")
    print(" ")
for i in range(4,0,-1):
    for j in range(i):
        print("*",end=" ")
    print(" ")

 
*  
* *  
* * *  
* * * *  
* * * * *  
* * * *  
* * *  
* *  
*  


**22.	Count Aplhabets,Numbers and special characters**

write a python program that accepts a string and calculate the number of digits and letters

    1
    12
    123
    1234
    12345
    1234
    123
    12
    1

In [113]:
for i in range(1,6):
    for j in range(1,i+1):
        print(j,end=" ")
    print(" ")
for i in range(4,0,-1):
    for j in range(1,i+1):
        print(j,end=" ")
    print(" ")

1  
1 2  
1 2 3  
1 2 3 4  
1 2 3 4 5  
1 2 3 4  
1 2 3  
1 2  
1  


**23. Write a python program to construct the following pattern**



In [120]:
for i in range(1,10):
    for j in range(i):
        print(i,end="")
    print("")

1
22
333
4444
55555
666666
7777777
88888888
999999999
